src/orchestrator-process/src/lib.rs

```diff
@@ -255,6 +258,35 @@ impl NamespacedOrchestrator for NamespacedProcessOrchestrator {
         let supervisors = self.supervisors.lock().expect("lock poisoned");
         Ok(supervisors.keys().cloned().collect())
     }
+
+    fn watch_services(&self) -> BoxStream<'static, Result<ServiceEvent, anyhow::Error>> {
+        let supervisors = Arc::clone(&self.supervisors);
+        let stream = async_stream::stream! {
+            let mut events = Vec::new();
+            loop {
+                {
+                    let supervisors = supervisors.lock().expect("lock poisoned");
+                    for (service, processes) in supervisors.iter() {
+                        for process_idx in 0..processes.len() {
+                            events.push(ServiceEvent {
+                                service_id: service.to_string(),
+                                process_id: process_idx as i64,
+                                status: ServiceStatus::Unknown,
```

Comment:
Yeah, we don't really care and getting a proper status is not trivial in this case. This was set to `Ready` initially, but then Philip pointed out that this was confusing because even crash-looping replicas showed as ready/healthy (see his Item 1 above). So I changed it to something more honest.